pkg/osbuild/source.go

```diff
@@ -54,25 +75,55 @@ func (sources *Sources) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
-func GenSources(packages []rpmmd.PackageSpec, ostreeCommits []ostree.CommitSpec, inlineData []string, containers []container.Spec) (Sources, error) {
+func addPackagesCurl(sources Sources, packages []rpmmd.PackageSpec) error {
+	curl := NewCurlSource()
+	for _, pkg := range packages {
+		err := curl.AddPackage(pkg)
+		if err != nil {
+			return err
+		}
+	}
+	sources["org.osbuild.curl"] = curl
```

Comment:
Nitpick: I'm wondering if now is a good time to define the source name as a constant in the osbuild package, so that we don't use literal strings everywhere. We could even have a special type for source names.

Reply:
I fixed this accross the entire file now, please have a look if it matches what you have in mind :)